staff/miguel-angel-zapatero/precourse/airlines-pro/index.js

```diff
@@ -0,0 +1,290 @@
+/* Después de ver toda la información el programa pedirá al usuario si es ADMIN/USER, dependiendo de la elección, el programa se comportará de la siguiente manera:
+
+Si eres ADMIN, la función debería permitir:
+- Poder crear, más vuelos, pidiendo la información por prompt(), sin poder pasar de 15 vuelos, si se intenta introducir uno más, saltará un alert().
+- Poder eliminar vuelos mediante el ID.
+
+Si eres USER la función debería permitir:
+ -Buscar por precio (más alto, más bajo o igual), el usuario debería mostrar los datos de los vuelos encontrados y, indicando el ID, el programa responderá: "Gracias por su compra, vuelva pronto." */
+
+let flights = [
+    { id: 00, to: 'Bilbao', from: 'Barcelona', cost: 1600, scale: false },
+    { id: 01, to: 'New York', from: 'Barcelona', cost: 700, scale: false },
+    { id: 02, to: 'Los Angeles', from: 'Madrid', cost: 1100, scale: true },
+    { id: 03, to: 'Paris', from: 'Barcelona', cost: 210, scale: false },
+    { id: 04, to: 'Roma', from: 'Barcelona', cost: 150, scale: false },
+    { id: 05, to: 'London', from: 'Madrid', cost: 200, scale: false },
+    { id: 06, to: 'Madrid', from: 'Barcelona', cost: 90, scale: false },
+    { id: 07, to: 'Tokyo', from: 'Madrid', cost: 1500, scale: true },
+    { id: 08, to: 'Shangai', from: 'Barcelona', cost: 800, scale: true },
+    { id: 09, to: 'Sydney', from: 'Barcelona', cost: 150, scale: true },
+    { id: 10, to: 'Tel-Aviv', from: 'Madrid', cost: 150, scale: false }
+];
+
+function showFlights(data, typeUser, find = false) {
+    if (data.length != 0) {
+        let allFlights = '';
+        // Podría cambiarlo para hacerlo mediante un forEach igual que en la función showScaleFlights.
+        for (let i = 0; i < data.length; i++) {
+            if (typeUser == 'ADMIN' || find == true) { //?!!!
+                allFlights += `ID: ${data[i].id} - `;
+            }
+            allFlights += `El vuelo con origen: ${data[i].from}, y destino: ${data[i].to} tiene un coste de ${data[i].cost}€`;
+            if (data[i].scale) {
+                allFlights += ' con escala.\n'
+            } else {
+                allFlights += ' y no realiza ninguna escala.\n'
+            }
+        }
+        return allFlights;
+    } else {
+        return 'No hay vuelos!' 
+    }
+}
+
+function showAvgCost(data) {
+    let sum = 0;
+    data.forEach(function(flight){
+        sum += flight.cost; 
+    });
+    return 'El coste medio de los vuelos es ' + (sum / data.length).toFixed(2) + '€';
```

Comment:
@maz-e as this code is right, you can improve this code by storing the average in a variable:

```
const average = (sum / data.length).toFixed(2);
return `El coste medio de los vuelos es ${average} €`;

```

Reply:
👍